src/code.cloudfoundry.org/silk-datastore-syncer/main.go

```diff
@@ -0,0 +1,180 @@
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerflags"
+	"code.cloudfoundry.org/lib/common"
+	"code.cloudfoundry.org/lib/datastore"
+	"code.cloudfoundry.org/lib/serial"
+)
+
+var (
+	interval      *int
+	gardenNetwork *string
+	gardenAddr    *string
+	silkFile      *string
+	silkFileOwner *string
+	silkFileGroup *string
+	logLevel      *string
+)
+
+func init() {
+	interval = flag.Int("n", 30, "sync interval in seconds.")
+	gardenNetwork = flag.String("gardenNetwork", "", "garden network type.")
+	gardenAddr = flag.String("gardenAddr", "", "garden address.")
+	silkFile = flag.String("silkFile", "", "silk file.")
+	silkFileOwner = flag.String("silkFileOwner", "", "owner of silk file")
+	silkFileGroup = flag.String("silkFileGroup", "", "group owner of silk file")
+	logLevel = flag.String("logLevel", lager.INFO.String(), "log level")
+}
+
+func main() {
+	flag.Parse()
+	loggerConfig := common.GetLagerConfig()
+	loggerConfig.LogLevel = *logLevel
+	logger, _ := lagerflags.NewFromConfig(fmt.Sprintf("%s.%s", "cfnetworking", "silk-datastore-syncer"), loggerConfig)
+
+	logger.Info("properties", lager.Data{
+		"interval":      interval,
+		"gardenNetwork": gardenNetwork,
+		"gardenAddr":    gardenAddr,
+		"silkFile":      silkFile,
+		"silkFileOwner": *silkFileOwner,
+		"silkFileGroup": *silkFileGroup,
+	})
+
+	gardenClient := client.New(connection.New(*gardenNetwork, *gardenAddr))
+	WaitForGarden(gardenClient, logger)
+	store := makeDatastore()
+
+	duration := time.Duration(*interval) * time.Second
+	for {
+		time.Sleep(duration)
+		logger.Debug("Starting sync loop")
+
+		gardenContainers, err := gardenClient.Containers(nil)
+		if err != nil {
+			logger.Error("Garden: error retrieving containers:", err)
+			continue
+		}
+
+		storeContainers, err := store.ReadAll()
+		if err != nil {
+			logger.Error("Datastore: error retrieving containers from datastore:", err)
+			continue
+		}
+
+		for _, c := range gardenContainers {
+			desiredLogConfig, err := getGardenLogConfig(c)
+			if err != nil {
+				logger.Error("error getting garden log config", err)
+				continue
+			}
+			logger.Debug("Garden container", lager.Data{"log info": desiredLogConfig})
+
+			sc := storeContainers[c.Handle()]
+			actualLogConfig, err := getSilkLogConfig(sc)
+			if err != nil {
+				logger.Error("error getting silk log config", err)
+				continue
+			}
+			logger.Debug("Datastore container", lager.Data{"log info": actualLogConfig})
+
+			if reflect.DeepEqual(desiredLogConfig, actualLogConfig) {
+				logger.Debug("They are equal no action taken")
+				continue
+			}
+
+			logger.Debug("Datastore container reconciling with Garden container", lager.Data{"handle": sc.Handle})
+			b, err := json.Marshal(desiredLogConfig)
+			if err != nil {
+				logger.Error("Garden container error marshalling container log config", err, lager.Data{"handle": sc.Handle})
+				continue
+			}
+			sc.Metadata["log_config"] = string(b)
+			err = store.Add(sc.Handle, sc.IP, sc.Metadata)
```

Comment:
This should maybe be a update. Could cause re-adding things on accident and leave limbo entries.

Reply:
from my perspective these tests are completely unrunnable tho as is. They just fail.